barman/backup_executor.py

```diff
@@ -1593,6 +1601,60 @@ def stop_backup(self, backup_info):
             # Skip switching XLOG if a standby server
             pass
 
+    @contextmanager
+    def _get_tar_fileobj(self, backup_info):
```

Comment:
This function is dealing with the fact that the tar file containing the backup_label needs to be opened using a specific compression type - I'm in two minds about whether this is the best place for that knowledge. On one hand, a backup strategy perhaps shouldn't need to care about specific compresion, but on the other hand I don't think there's currently a better place for this knowledge to go.